docs/docs/production-deployment.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Production Deployment
+description: >-
+ This article covers production deployment requirements and concerns
+meta:
+  - name: keywords
+    content: pomerium, identity access proxy, production, deployment, scale, scaling, horizontal,
+---
+
+
+# Production Deployment
+
+This page covers the topic of running Pomerium in a production configuration.
+
+Before deploying Pomerium to Production, you should have already tested Pomerium in one or more demo environments and confirmed:
+
+ - A working connection to your [IdP](/docs/glossary.md#identity-provider).
+ - Working test routes to your upstream services, including [JWT verification] where applicable.
+ - For Pomerium Enterprise, a working demo of the Pomerium Enterprise Console, with confirmed access for your [Admins and Managers][rbac].
+
+
+## Service Mode
+
+Pomerium is designed to be run in two modes: All-In-One or Split Service. These modes are not mutually exclusive, meaning you can run one or multiple instances of Pomerium in all-in-one mode, and spin up additional instances for specific components as needed.
+
+Each instance of Pomerium runs in all-in-one mode unless specified to run as a specific component by the `services` key. See [All-In-One vs Split Service mode](/reference/readme.md#all-in-one-vs-split-service-mode) for more details.
+
+It's important to note that any production deployment with more than one instance of Pomerium (in any combination of modes) should be configured to use Redis as the [`databroker_storage_type`](/reference/readme.md#data-broker-storage-type). See [Data Storage - Backends](/docs/topics/data-storage.md#backends) for more details.
+
+### All in One
+
+In smaller deployments or while testing, it may be desirable to run in "all-in-one" mode. This reduces the resource footprint and simplifies DNS configuration. All URLs point at the same pomerium service instance.
+
+### Discrete Services
+
+In larger footprints, it is recommended to run Pomerium as a collection of discrete service clusters. This limits blast radius in the event of vulnerabilities and allows for per-service [scaling](#scaling) and monitoring.
+
+Please also see [Architecture](/docs/architecture.md) for information on component interactions.
+
+## Scaling
+
+In split service mode, you have the opportunity to scale the components of Pomerium independently.
+
+All of Pomerium's components are designed to be stateless, and may all be scaled horizontally or vertically. In general, horizontal scaling is recommended. Vertical scaling will lead to diminished returns after ~8 vCPUs.
```

Comment:
just a comment here that it might be a good idea to add stateless to glossary, since I have no idea what it means